Web Inspector: Replace MessageSource::AppCache in WebCore

rcaliman-apple opened the pull request:
#### b3fe8b76f0523718b025d4f80c03663a996506c1
<pre>
Web Inspector: Replace MessageSource::AppCache in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=294907">https://bugs.webkit.org/show_bug.cgi?id=294907</a>
<a href="https://rdar.apple.com/154202768">rdar://154202768</a>

Reviewed by Devin Rousso.

ApplicationCache (aka AppCache) is a deprecated feature:
<a href="https://bugs.webkit.org/show_bug.cgi?id=181764">https://bugs.webkit.org/show_bug.cgi?id=181764</a>

Support for inspecting AppCache was removed from WebInspector:
<a href="https://commits.webkit.org/289820@main">https://commits.webkit.org/289820@main</a>

With the feature is still technically available in WebCore,
it can emit some console messages on an AppCache-specific message channel.

WebInspector frontend doesn&apos;t explicitly handle them anymore.

This patch replaces the `MessageSource::AppCache` with `MessageSource::Other`
to ensure that any potential messages still get handled.

* Source/JavaScriptCore/inspector/ConsoleMessage.cpp:
(Inspector::messageSourceValue):
* Source/JavaScriptCore/inspector/protocol/Console.json:
* Source/JavaScriptCore/runtime/ConsoleClient.cpp:
(JSC::appendMessagePrefix):
* Source/JavaScriptCore/runtime/ConsoleTypes.h:
* Source/WebCore/inspector/agents/page/PageConsoleAgent.cpp:
(WebCore::PageConsoleAgent::getLoggingChannels):
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
(WebCore::ApplicationCacheGroup::abort):
(WebCore::ApplicationCacheGroup::didFinishLoadingEntry):
(WebCore::ApplicationCacheGroup::didFailLoadingEntry):
(WebCore::ApplicationCacheGroup::didFinishLoadingManifest):
(WebCore::ApplicationCacheGroup::didFailLoadingManifest):
(WebCore::ApplicationCacheGroup::checkIfLoadIsComplete):
* Source/WebKit/Shared/JavaScriptCore.serialization.in:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(stringForMessageSource):
* Source/WebKitLegacy/mac/WebKit.exp:
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:

Canonical link: <a href="https://commits.webkit.org/296659@main">https://commits.webkit.org/296659@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e1b61af8fa8b366abf7c10c8ca76c98f3a93c3bb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108933 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28593 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19018 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114141 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59275 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110896 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29277 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37160 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82779 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111881 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23276 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98119 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63218 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22697 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16256 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58839 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/101470 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92650 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16303 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117261 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/107492 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35983 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26593 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91788 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36355 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94383 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91595 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23382 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36504 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14261 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31850 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35881 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41408 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/131767 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35582 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35728 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38923 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37266 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
```diff
@@ -144,7 +144,6 @@ extern NSString *WebConsoleMessageJSMessageSource;
 extern NSString *WebConsoleMessageNetworkMessageSource;
 extern NSString *WebConsoleMessageConsoleAPIMessageSource;
 extern NSString *WebConsoleMessageStorageMessageSource;
-extern NSString *WebConsoleMessageAppCacheMessageSource;
```
we may need to keep this if there's a client elsewhere at Apple that still uses it

rcaliman-apple replied:
I hesitated making changes under `Source/WebKitLegacy/`. 
@cdumez, do we still touch that or should I leave it unchanged in this patch?

rcaliman-apple replied:
> we may need to keep this if there's a client elsewhere at Apple that still uses it

Nobody seems to be using this anymore. 

If they somehow are and hit a crash, it will be a good way to identify themselves by filing a bug so we can ask them to migrate (or revert this change if needed).

rcaliman-apple committed:
Web Inspector: Replace MessageSource::AppCache in WebCore
https://bugs.webkit.org/show_bug.cgi?id=294907
rdar://154202768

Reviewed by Devin Rousso.

ApplicationCache (aka AppCache) is a deprecated feature:
https://bugs.webkit.org/show_bug.cgi?id=181764

Support for inspecting AppCache was removed from WebInspector:
https://commits.webkit.org/289820@main

With the feature is still technically available in WebCore,
it can emit some console messages on an AppCache-specific message channel.

WebInspector frontend doesn't explicitly handle them anymore.

This patch replaces the `MessageSource::AppCache` with `MessageSource::Other`
to ensure that any potential messages still get handled.

* Source/JavaScriptCore/inspector/ConsoleMessage.cpp:
(Inspector::messageSourceValue):
* Source/JavaScriptCore/inspector/protocol/Console.json:
* Source/JavaScriptCore/runtime/ConsoleClient.cpp:
(JSC::appendMessagePrefix):
* Source/JavaScriptCore/runtime/ConsoleTypes.h:
* Source/WebCore/inspector/agents/page/PageConsoleAgent.cpp:
(WebCore::PageConsoleAgent::getLoggingChannels):
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
(WebCore::ApplicationCacheGroup::abort):
(WebCore::ApplicationCacheGroup::didFinishLoadingEntry):
(WebCore::ApplicationCacheGroup::didFailLoadingEntry):
(WebCore::ApplicationCacheGroup::didFinishLoadingManifest):
(WebCore::ApplicationCacheGroup::didFailLoadingManifest):
(WebCore::ApplicationCacheGroup::checkIfLoadIsComplete):
* Source/WebKit/Shared/JavaScriptCore.serialization.in:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(stringForMessageSource):
* Source/WebKitLegacy/mac/WebKit.exp:
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:

Canonical link: https://commits.webkit.org/296659@main
